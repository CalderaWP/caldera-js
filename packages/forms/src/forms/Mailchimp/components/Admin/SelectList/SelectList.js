import React, {Fragment} from 'react'
import {Field} from '@calderajs/components';

/**
 * Select a MailChimp List
 *
 * @param listFieldConfig
 * @param listId
 * @param setList
 * @param instanceId
 * @return {*}
 * @constructor
 */
export const SelectList = ({listFieldConfig,listId,setListId,instanceId}) => (
	<Fragment>
		<Field
			field={{
				...listFieldConfig,
				fieldId: `caldera-mc-select-list-${instanceId}`,
				value: listId,
			}}
			onChange={(newValue)=>{
				setListId(newValue);
			}}
		/>
	</Fragment>
);
