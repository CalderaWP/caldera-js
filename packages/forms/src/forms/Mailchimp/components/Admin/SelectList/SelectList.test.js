import React from 'react';
import { mount } from 'enzyme/build';
import { render, fireEvent, getByTestId } from 'react-testing-library';
import { SelectList } from './SelectList';

const field = [
	{
		fieldType: 'select',
		required: true,
		fieldId: 'mc-select-field',
		options: [{ value: '45907f0c59', label: 'Future Capable' }],
	},
];
it('Shows list', () => {
	expect(
		render(
			<SelectList
				listFieldConfig={field}
				listId={'45907f0c59'}
				setList={() => {}}
				instanceId={'a'}
			/>
		)
	).toMatchSnapshot();
});

it('Changes list', () => {
	const onChange = jest.fn();
	const component = mount(
		<SelectList
			listFieldConfig={field[0]}
			listId={''}
			setListId={onChange}
			instanceId={'1'}
		/>
	);

	component
		.find('select')
		.first()
		.simulate('change', { target: { value: '45907f0c59' } });
	expect(onChange.mock.calls.length).toBe(1);
});
