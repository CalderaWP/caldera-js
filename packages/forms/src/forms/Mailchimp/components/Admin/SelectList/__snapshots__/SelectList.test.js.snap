// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shows list 1`] = `
<div
  className="caldera-field-wrapper caldera-field-wrapper-text"
>
  <label
    className="caldera-field-label caldera-field-label-text"
    htmlFor="caldera-mc-select-list-a"
  >
     
  </label>
  <input
    className="caldera-field caldera-field-text"
    id="caldera-mc-select-list-a"
    onBlur={[Function]}
    onChange={[Function]}
    type="text"
    value="45907f0c59"
  />
</div>
`;
